sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/FieldBuilder.java

```diff
@@ -0,0 +1,268 @@
+        Field searchField = new Field();
+        searchField.setName(classField.getName());
+        DataType dataType = covertToDataType(classField.getGenericType(), false, logger);
+        searchField.setType(dataType);
+        searchField.setKey(false);
+        searchField.setSearchable(false);
+        searchField.setFacetable(false);
+        searchField.setHidden(false);
+        searchField.setFilterable(false);
+        searchField.setSortable(false);
+        return searchField;
+    }
+
+    private static Field enrichWithAnnotation(Field searchField, java.lang.reflect.Field classField,
+        ClientLogger logger) {
+        if (classField.isAnnotationPresent(SimpleFieldProperty.class)
+            && classField.isAnnotationPresent(SearchableFieldProperty.class)) {
+            throw logger
+                .logExceptionAsError(new IllegalArgumentException(
+                    String.format("@SimpleFieldProperty and @SearchableFieldProperty cannot be present simultaneously for %s",
+                        classField.getName())));
+        }
+        if (classField.isAnnotationPresent(SimpleFieldProperty.class)) {
+            SimpleFieldProperty simpleFieldPropertyAnnotation =
+                classField.getDeclaredAnnotation(SimpleFieldProperty.class);
+            searchField.setSearchable(false);
+            searchField.setSortable(simpleFieldPropertyAnnotation.isSortable());
+            searchField.setFilterable(simpleFieldPropertyAnnotation.isFilterable());
+            searchField.setFacetable(simpleFieldPropertyAnnotation.isFacetable());
+            searchField.setKey(simpleFieldPropertyAnnotation.isKey());
+            searchField.setHidden(simpleFieldPropertyAnnotation.isHidden());
+        } else if (classField.isAnnotationPresent(SearchableFieldProperty.class)){
+            if (!searchField.getType().equals(DataType.EDM_STRING) &&
+                !searchField.getType().equals(DataType.collection(DataType.EDM_STRING))) {
+                throw logger.logExceptionAsError(new RuntimeException(String.format("SearchFieldProperty can only"
+                    + " be used on string properties. Property %s returns a %s value.",
+                    classField.getName(), searchField.getType())));
+            }
+            SearchableFieldProperty searchableFieldPropertyAnnotation =
+                classField.getDeclaredAnnotation(SearchableFieldProperty.class);
+            searchField.setSearchable(true);
+            searchField.setSortable(searchableFieldPropertyAnnotation.isSortable());
+            searchField.setFilterable(searchableFieldPropertyAnnotation.isFilterable());
+            searchField.setFacetable(searchableFieldPropertyAnnotation.isFacetable());
+            searchField.setKey(searchableFieldPropertyAnnotation.isKey());
+            searchField.setHidden(searchableFieldPropertyAnnotation.isHidden());
+            if (!"null".equals(searchableFieldPropertyAnnotation.analyzer())) {
```

Comment:
Should just be a simple `null` check if not defined as a string, right? What if they set it to `null` explicitly when declaring?

Reply:
Switch to empty